### ISSUE TITLE
Codegen member objects naming convention

### ISSUE THREAD

**Issue by axcsrs** [qcubed/qcubed#1307]
I have, for example, a table Awards that has a foreign key to table AwardType.
In v2.x, when you had a foreign key column in a table that ended in _id (fk_award_type_id) and referenced a primary key that ended in _id (pk_award_type_id), the codegen generated property names would be something like $objFkAwardType.  If the columns were named differently, you'd get $objFKAWARDTYPEObject.  In v3.1.0, I am only getting property names like the latter (which is an issue because I am trying to upgrade a v2-based application).   Is there a setting I am missing, or someplace in the codegen templates I can change this behavior?

**Comment by matthiaz** [comment]
You can overwrite each codegen template as you please.

There are codegen templates in framework's composer directory, but if you
want to overwrite them, you can simply copy them to your project folder.
There is a codegen template folder there as well.
It takes a bit of searching to figure out which file generates what, so I
usually just do a find on what I see in the resulting gens, to figure out
what to change and where.

2018-06-12 18:15 GMT+02:00 Adriana C <notifications@github.com>:

**Comment by spekary** [comment]
It sounds like it is not interpretting the type table as a type table, but rather as a regular object. Make sure your type table name ends with "_type" and that you haven't changed the type table ending in your codegen_settings.xml file.

**Comment by axcsrs** [comment]
Sorry, that was a bad example.  These are not necessarily type tables, these are any associated member object.  So it would include Awards->Departments and Awards->People, which I used to be able to access via $objAwards->FkAwardsHomeDepartment and $objAwards->FKPiPeople instead of $objAwards->FKAWARDSHOMEDEPTIDObject and $objAwards->FKPIPEOPLEIDObject.  

I found the responsible code, in v2.3.0, it is QCodeGenBase ReferenceColumnNameFromColumn, line ~849, which is ModelReferenceColumnName in v3.1.0, line ~646.  It looks like the comparison in v3 is missing a strtolower.  I will try that & report back whether it fixes the issue.
